src/components/progress/progress.ts

```diff
@@ -0,0 +1,28 @@
+import { Component, Input, ViewEncapsulation, HostBinding } from '@angular/core';
+
+@Component({
+  selector: 'ion-progress',
+  templateUrl : './progress.html',
+  encapsulation: ViewEncapsulation.None
+})
+export class Progress {
+
+  constructor( ) { }
+
+  /**
+   * @input {'determinate' | 'indeterminate' | 'buffer' | 'query'} The current mode of the progress indicator
```

Comment:
add (above the class): `export type progressIndicator = 'determinate' | 'indeterminate' | 'buffer' | 'query'` and use it instead of explicit strings